shoop/core/utils/shipments.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+
+
+def generate_shipment_key(order):
```

Comment:
No, it should be in `Shipment.__init__`, so that unsaved Shipments will have identifiers too. Init should generate an identifier if one is not specified in kwargs.